src/NuGet.Core/NuGet.Packaging/PackagesConfigWriter.cs

```diff
@@ -451,6 +485,26 @@ public void Close()
             }
         }
 
+        /// <summary>
+        /// Write the XDocument to the packages.config and disallow further changes.
+        /// </summary>
+        /// <param name="fullPath">the full path to packages.config file</param>
+        public void Close(string fullPath)
+        {
+            var directorypath = Path.GetDirectoryName(fullPath);
+
+            var tempConfigPath = Path.Combine(directorypath, Path.GetRandomFileName());
+
+            using (var tempConfigStream = File.Create(tempConfigPath))
+            {
+                // Save the XDocument to the temp stream
+                _xDocument.Save(tempConfigStream);
+            }
+
+            File.Copy(tempConfigPath, fullPath, true);
+            File.Delete(tempConfigPath);
```

Comment:
File.Move will throw an exception when the targetFilePath already exists.

I can probably do a File.Delete of the existing packages.config file and then do a File.Move